devito/operator.py

```diff
@@ -96,6 +96,17 @@ def expr_cse(expr):
                     s_dict[arg] = Indexed(to_revert[value.base.label], *value.indices)
         to_revert[temp] = value.xreplace(s_dict)
 
+    # Make sure that no temporary that has to be reverted depends
+    # on another temporary of the list
+    changing = True
+    while changing:
+        changing = False
+        for temp, value in to_revert.items():
+            n_value = value.xreplace(to_revert)
```

Comment:
So I will have to see with gerard, I am not allowed to put the example on opesci, but mainly the error raises when :
` temp1 = temp2 - temp3 * tem4[temp5, temp6]`
I get then a `TypeError` on `xreplace`